File: LeetCode/1700. Number of Students Unable to Eat Lunch/1700.java
class Solution {
    public int countStudents(int[] students, int[] sandwiches) {
        Queue<Integer> queue = new LinkedList<>();
        int[] count = new int[2];
        for(int student:students){
            queue.offer(student);
            count[student]++;
        }

        int i=0;
        while(!queue.isEmpty()&&count[sandwiches[i]]>0){
            if (queue.peek()==sandwiches[i]){
                count[queue.poll()]--;
                i++;
            }else{
                queue.offer(queue.poll());
            }
        }
        return queue.size();
    }
}

File: LeetCode/1700. Number of Students Unable to Eat Lunch/1700.py
class Solution(object):
    def countStudents(self, students, sandwiches):
        """
        :type students: List[int]
        :type sandwiches: List[int]
        :rtype: int
        """
        queue = deque(students)  # Öğrencileri bir kuyrukta saklıyoruz.
        count = [0, 0]  # Her sandviç türü için tercih sayısını tutuyoruz.
        
        # Öğrenci tercihlerini sayıyoruz.
        for student in students:
            count[student] += 1
        
        i = 0
        while queue and count[sandwiches[i]] > 0:
            if queue[0] == sandwiches[i]:
                # Eğer kuyruğun başındaki öğrenci sandviçi yiyebiliyorsa
                count[queue.popleft()] -= 1
                i += 1
            else:
                # Eğer yiyemiyorsa, öğrenciyi kuyruğun sonuna ekliyoruz.
                queue.append(queue.popleft())
        
        # Yiyemeyen öğrencilerin sayısını döndürüyoruz.
        return len(queue)
